.github/workflows/test.yml

```diff
@@ -57,8 +57,9 @@ jobs:
           if-no-files-found: ignore
 
       - name: Upload coverage report
-        if: ${{ always() && !cancelled() }}
```

Comment:
Why remove this line?

If anything I would keep it but slightly modified:

```
if: ${{ !cancelled() && always() }}
```

this way it'll only upload the coverage report if the workflow isn't cancelled...